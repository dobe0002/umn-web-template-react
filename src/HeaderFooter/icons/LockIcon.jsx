import React from 'react';
import PropTypes from 'prop-types';

const LockIcon = (props) => {
  const { color } = props;
  return (
    <svg
      width='9px'
      height='12px'
      viewBox='0 0 9 12'
      version='1.1'
      xmlns='http://www.w3.org/2000/svg'
    >
      <g stroke='none' strokeWidth='1' fill='none' fillRule='evenodd'>
        <g fill={color} fillRule='nonzero'>
          <path d='M0,5.4 L0,12 L9,12 L9,5.4 M0.642857143,5.4 L0.642857143,3.6 C0.642857143,1.62 2.37857143,0 4.5,0 C6.62142857,0 8.35714286,1.62 8.35714286,3.6 L8.35714286,5.4 L7.07142857,5.4 L7.07142857,3.6 C7.07142857,2.28 5.91428571,1.2 4.5,1.2 C3.08571429,1.2 1.92857143,2.28 1.92857143,3.6 L1.92857143,5.4' />
        </g>
      </g>
    </svg>
  );
};

export default LockIcon;

LockIcon.propTypes = {
  color: PropTypes.string,
};
LockIcon.defaultProps = {
  color: '#fff',
};
