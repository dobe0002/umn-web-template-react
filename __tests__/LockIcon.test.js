import React from 'react';
import { configure, shallow } from 'enzyme';
import LockIcon from '../src/HeaderFooter/LockIcon';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('LockIcon', () => {
  it('LockIcon renders correctly', () => {
    const lockIcon = shallow(<LockIcon />);
    expect(lockIcon).toMatchSnapshot();
  });
});
