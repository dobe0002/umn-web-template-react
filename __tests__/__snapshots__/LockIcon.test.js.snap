// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LockIcon LockIcon renders correctly 1`] = `
<svg
  height="12px"
  version="1.1"
  viewBox="0 0 9 12"
  width="9px"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    fill="none"
    fillRule="evenodd"
    id="Page-1"
    stroke="none"
    strokeWidth="1"
  >
    <g
      fill="#fff"
      fillRule="nonzero"
    >
      <path
        d="M0,5.4 L0,12 L9,12 L9,5.4 M0.642857143,5.4 L0.642857143,3.6 C0.642857143,1.62 2.37857143,0 4.5,0 C6.62142857,0 8.35714286,1.62 8.35714286,3.6 L8.35714286,5.4 L7.07142857,5.4 L7.07142857,3.6 C7.07142857,2.28 5.91428571,1.2 4.5,1.2 C3.08571429,1.2 1.92857143,2.28 1.92857143,3.6 L1.92857143,5.4"
        id="Shape"
      />
    </g>
  </g>
</svg>
`;
